cupy/cuda/__init__.py

```diff
@@ -33,6 +34,14 @@
 except ImportError:
     thrust_enabled = False
 
+cub_enabled = False
+if int(os.getenv('CUB_DISABLED', 0)) == 0:
```

Comment:
I'm a bit confused, so let me check.
You think that we should stop using the environment variable CUB_DISABLED and switch back to using the function use_cub(), right?

Reply:
> This is contrary to this comment, 

This is because the code snippet there was opposite to his intention :)

Reply:
Is the following implementation OK?

1. If CuPy is NOT build with CUB
        --> CUB is never used.
1. If CuPy is build with CUB
   1. If an environment variable `CUB_DISABLED` is set
        --> CUB is never used.
   1. If an environment variable `CUB_DISABLED` is not set
        --> CUB is used as default.
      1. If a function `disable_cub()` is called.
        --> CUB is not used as default.
      1. If a function `enable_cub()` is called.
        --> CUB is used as default.

Reply:
Sorry for being unclear.
We don't need an interface for users to switch the behavior in this PR.
I think the previous implementation was better because after #2436 is implemented CuPy will need to determine the behavior at runtime. If `cupy.cuda.cub_enabled` is only for internal use, we don't even have to do that now. We can just implement that later.

Reply:
All right. So, you are suggesting to revert the last commit (https://github.com/cupy/cupy/pull/2090/commits/88d6a8d1215971ceec42f24e00ac7aa608fc39fa), right?

Reply:
Yes, that's what I meant. But I'll merge the PR now anyway as that can be done later.